### ISSUE TITLE
Upgrade to 2.3.0 causes endless "Make sure `sst dev` is running..." when running tests

### ISSUE THREAD

**Issue by bobbypiper** [serverless-stack/sst#2734]
We're currently running a few minor versions behind at 2.1.35; I got round to upgrading today, but can go no further than v2.2.8 
 (I did a sort of manual bisect between 2.1.35 and 2.3.6).

I think this block is the problem child in v2.3.0 - https://github.com/serverless-stack/sst/compare/v2.2.8...v2.3.0#diff-d63c9e7a9863532ee3679a6f3df0609ae68a24bf951f93caa1f04128bfbc90f6R203-R210

now when running `pnpm test`  we get `Make sure sst dev is running...` - the odd thing is, even with `pnpm dev` running the tests still do not run and the above message is as far as I get.

When running a ripgrep against `sst` I can find the running `dev` process:

![image](https://user-images.githubusercontent.com/1521472/229745855-d8684503-6d3b-4cf7-835d-6b0bdc564cee.png)

Also; if it's now a requirements to run `sst dev` before running tests it might be worth updating the testing docs https://docs.sst.dev/testing

**Comment by bobbypiper** [comment]
On closer look I think I've spotted the problem.

https://github.com/serverless-stack/sst/compare/v2.2.8...v2.3.0#diff-d63c9e7a9863532ee3679a6f3df0609ae68a24bf951f93caa1f04128bfbc90f6R197-R200
```ts
// Do not retry if not running in frontend
if (!data && !isFrontend) {
  return;
}
```

The `isFrontend` var calls `isRunningInFrontend`

https://github.com/serverless-stack/sst/compare/v2.2.8...v2.3.0#diff-d63c9e7a9863532ee3679a6f3df0609ae68a24bf951f93caa1f04128bfbc90f6R100-R112

```ts
async function isRunningInFrontend() {
  const { existsAsync } = await import("../../util/fs.js");
  const results = await Promise.all(
    Object.values(SITE_CONFIG)
      .map((config) =>
        [".js", ".cjs", ".mjs", ".ts"].map((ext) =>
          existsAsync(`${config}${ext}`)
        )
      )
      .flat()
  );
  return results.some(Boolean);
}
```
Which looks to see if any of the specified config files exist:

```ts
const SITE_CONFIG = {
  NextjsSite: "next.config",
  AstroSite: "astro.config",
  RemixSite: "remix.config",
  SolidStartSite: "vite.config",
  StaticSite: "vite.config",
  SlsNextjsSite: "next.config",
};
```

As we're using https://vitest.dev/ for our tests  we will have a `vite.config` present - but the application isn't frontend, it's a collection of integrated AWS resources for background processing.

**Comment by fwang** [comment]
Fixed in v2.4.1
